test/e2e/providers/shared/shared.go

```diff
@@ -0,0 +1,74 @@
+package shared
```

Comment:
Rather than `shared` I suggest calling this `machineset` given that is what it is dealing with.